### PR TITLE
Add Customer Add Command

### LINKED ISSUE #11: As a registered owner, I can create a customer
...so that I can efficiently organize and access customer information

### PR DESCRIPTION
**Pull request by zhonghan721**
Close #11 

Update add command to align with UG
Update JSON related classes to support reading and writing of customerId
~~Revert updating JSON classes to write customerId into data file~~
~~Modify customerId to serve as display index to align with other commands~~

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Customer.java [hunk 1]
```diff
@@ -11,13 +11,13 @@
...
 /**
- * Represents a Person in the address book.
+ * Represents a Customer in the address book.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Customer {
 
     // Customer ID Generation
-    private static int customerCount;
+    private static int customerCount = 1;
```
Comment (juliusgambe) [inline src/main/java/seedu/address/model/person/Customer.java:20]: Nice one!

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 2]
```diff
@@ -103,7 +107,22 @@ public Customer toModelType() throws IllegalValueException {
         final Address modelAddress = new Address(address);
 
         final Set<Tag> modelTags = new HashSet<>(personTags);
-        return new Customer(modelName, modelPhone, modelEmail, modelAddress, modelTags);
+
+        // For customers without customerId
+        if (customerId == null) {
+            return new Customer(modelName, modelPhone, modelEmail, modelAddress, modelTags);
+        }
```
Comment (juliusgambe) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:114]: Hmm would there be a case where `customerId == null`? or is this a workaround for the placeholder data. 
cc: @jianyangg @B-enguin @Gabriel4357

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 3]
```diff
@@ -47,12 +47,15 @@ public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
...
+        int maxCustomerId = 0;
         for (JsonAdaptedPerson jsonAdaptedPerson : persons) {
             Customer customer = jsonAdaptedPerson.toModelType();
             if (addressBook.hasPerson(customer)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_PERSON);
             }
             addressBook.addPerson(customer);
+            maxCustomerId = Math.max(maxCustomerId, customer.getCustomerId());
+            Customer.setCustomerCount(maxCustomerId);
         }
```
Comment (juliusgambe) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:59]: Good one!

**Comment by juliusgambe** [comment]
LGTM

**Review by juliusgambe — APPROVED** [review]
LGTM

**Review by Gabriel4357 — COMMENTED** [review]
Maybe could rename AddCommand to CustomerAddCommand for standardisation but other than that LGTM

**Pull request merged**